3_javascript/javascript1/week3/script.js

```diff
@@ -0,0 +1,159 @@
+
+//Item array removal
+
+const names = [
+    "Peter",
+    "Ahmad",
+    "Yana",
+    "kristina",
+    "Rasmus",
+    "Samuel",
+    "katrine",
+    "Tala",
+  ];
+  const nameToRemove = "Ahmad";
+  
+  const indexToRemove = names.indexOf(nameToRemove);
+
+  if (indexToRemove !== -1) {
+    names.splice(indexToRemove, 1)
+  }
+  
+  console.log(names); 
+
+
+//When will we be there??
+
+const travelInformation = {
+    speed: 50,
+    destinationDistance: 432,
+  };
+
+  function calculateTime (travelInformation){
+    const { speed, destinationDistance } = travelInformation;
+    
+    const totalTimeInHours = destinationDistance / speed;
+
+    const hours = Math.floor(totalTimeInHours);
+    const minutes = Math.round((totalTimeInHours - hours) * 60);
+
+    const formattedTime = `${hours} hours and ${minutes} minutes`;
+
+    return formattedTime;
+
+  }
+  
+  const travelTime = calculateTime(travelInformation);
+  console.log(travelTime); 
+
+  //Series duration of my life
+
+  const seriesDurations = [
+    {
+      title: "Game of thrones",
+      days: 3,
+      hours: 1,
+      minutes: 0,
+    },
+    {
+      title: "Sopranos",
+      days: 3,
+      hours: 14,
+      minutes: 0,
+    },
+    {
+      title: "The Wire",
+      days: 2,
+      hours: 12,
+      minutes: 0,
+    },
+  ];
+  
+  function calculateMinutes(seriesDurations) {
+    let totalminutes = 0;
+
+    for ( const series of seriesDurations){
+      const {days, hours, minutes} = series;
+      totalminutes = days * 24 * 60 + hours * 60 + minutes; 
+    }
+    return totalminutes;
+  }
+
+    function calculatePercentageOfLife(averageLife , totalminutes){
+      const minutesInYear = 365 * 24 * 60;
+      const percentage = (totalminutes / (averageLife * totalminutes) * 100 );
+      return percentage.toFixed(3);
+    }
+
+    function logOutSeriesText (){
+      const minutesWatchedSeries = calculateMinutes(seriesDurations);
+      const averageLife = 80;
+
+      let seriesOutput = "";
+
+      for (const series of seriesDurations){
+        const {title, days, hours, minutes} = series;
+        const seriesMinutes = days * 24 * 60 + hours * 60 + minutes; 
+        const percentageOfLife = calculatePercentageOfLife(seriesMinutes, averageLife)
+        seriesOutput += `${title} took ${percentageOfLife} in my life\n`
+
+      }
+
+      const totalPercentage = calculatePercentageOfLife(minutesWatchedSeries, averageLife);
+
+      console.log(seriesOutput);
+      console.log(`Games of Throne took ${totalPercentage}% of my life`)
+
+    }
+  
+  logOutSeriesText();
+
```

Comment:
🌟 Fantastic effort on this task!

👍 Your approach of splitting the functionality into separate functions is commendable. This helps in readability and maintainability of the code.

✨ Suggestions:

In the `calculatePercentageOfLife` function, it seems there might be a misunderstanding in the formula. You're dividing by `averageLife * totalminutes` which doesn't seem right. Consider revising it.

Instead of hardcoding "Games of Throne" in the final log message, it would be better to say "All series took...", as the `seriesDurations` array could be updated or changed, and this hardcoded message may not be valid anymore.

Instead of concatenating the output to `seriesOutput`, you can directly log out the message within the loop. This way, you won't have to accumulate all the messages in a single string.